docs/collisions.md

```diff
@@ -64,6 +64,62 @@ For example if a folder has five 5 with the same name say **_A.txt _** the five
 * [Extensions in folder names matter](https://apple.stackexchange.com/questions/123001/renamed-folder-becomes-a-file-with-an-extension) and should be handled :
 * Case sensitivity is taken into account. `Myfile` and `myFile` are different for example. That is useful to [ensure complete syncs](https://rclone.org/overview/#case-insensitive).
 
+#### Extensions
+
+For both files and folders the extension is any string that follows after the last dot.
+
+For example:
+<table>
+   <thead>
+      <th>File/Folder Name</th>
+      <th>Extension</th>
+      <th>Remark</th>
+   </thead>
+   <tbody>
+      <tr>
+         <td>My File.txt</td>
+         <td>txt</td>
+         <td>Regular</td>
+      </tr>
+      <tr>
+         <td>My File.txt(1)</td>
+         <td>txt(1)</td>
+         <td>Extension contains non-alphanumeric chars</td>
+      </tr>
+      <tr>
+         <td>My File.app</td>
+         <td>app</td>
+         <td>Regular</td>
+      </tr>
+      <tr>
+         <td>something.something</td>
+         <td>something</td>
+         <td>Name same as extension</td>
+      </tr>
+      <tr>
+         <td>2022.03.02</td>
+         <td>02</td>
+         <td>Extension is number</td>
+      </tr>
+      <tr>
+         <td>Folder.Work Files</td>
+         <td>Work Files</td>
+         <td>Extension contains spaces</td>
+      </tr>
+   </tbody>
+</table>
+
+As seen, extension can take infinite number of "shapes" BUT as long as all characters in the file name including the extension are valid file name chacters, then, **the extension is the string that follows after the last dot** (Emphasis).
+
+
+#### Source of extensions
+
+a) The **source string for file extensions ON FILES** could be the **`displayName`** OR the **`uploadFileName`**. When a file is uploaded to Permanent if the extension is recognized, then the `displayName` is calculated as `uploadFileName - File Extension`. So, if `My File.txt` is uploaded the `displayName` is `My File` while the `uploadFileName` would be `My File.txt`. This means when Permanent recognizes the extension, it removes it (the extension) from the `displayName` and keeps track of it(the extension) via the `uploadFileName`. When Permanent does not recognize the extension, the `uploadFileName` is thesame as `displayName` **at the point of creation**. For exampl, in the case of `File My.abcxyz` both the `uploadFileName` and `displayName` would be `My File.abcxyz` and so in this case the source of the extension is the `displayName` even though the extension can be obtained from the `uploadFileName` it risk being wrong if the user ever updates the `displayName`. This is the case because an update to the `displayName` does not affect the `uploadFileName` correctly so.
```

Comment:
Best to explain what "recognized" means in this context.  And I think you're using " - " as a minus sign -- that is, subtracting the extension from the `uploadFileName`.  But it's easy to read it as a literal space-hyphen-space.  So it's better to write it out in words: "...`displayName` is calculated as `uploadFileName` minus the extension. ..."

You can avoid an awkward parenthetical by avoiding pronouns in the first place: "...This means when Permanent recognizes the extension, we don't include the extension in `displayName` but we keep track of the extension via `uploadFileName`. ..."


Typo: "thesame" should be "the same".

Another typo: "exampl".  (Suggestion: run changes through an automated spell-checker before committing!  I do that all the time with documentation commits, FWIW.)

Now, the documentation I am reading here is obsolete, right, because we're just always going to use `uploadFileName` to get the extension, because the extension is, by definition, always available there?

Reply:
Resolved!